ui/components/component-library/popover/README.mdx

```diff
@@ -0,0 +1,265 @@
+<Popover position={PopoverPosition.LeftStart}>LeftStart</Popover>
+<Popover position={PopoverPosition.LeftEnd}>LeftEnd</Popover>
+```
+
+### Is Portal
+
+The `isPortal` prop is a boolean that when set to true, causes the Popover to be rendered as a separate DOM element at the end of the document body.
+Default `true`
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover hasArrow>Popover with arrow</Popover>;
+```
+
+### Has Arrow
+
+Use the `hasArrow` boolean to add an arrow to the popover.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover hasArrow>Popover with arrow</Popover>;
+```
+
+### Is Open
+
+Use the `isOpen` boolean to control the visibility of the popover.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover isOpen={true}>Popover with arrow</Popover>;
+```
+
+### Flip
+
+Use the `flip` boolean to flip the popover to the opposite side of the reference element if there is not enough space.
+For `PopoverPosition.Auto` this will become true.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover flip={true}>Flip demo</Popover>;
+```
+
+### Prevent Overflow
+
+Use the `preventOverflow` boolean to prevent the popover from overflowing the viewport.
+For `PopoverPosition.Auto` this will become true.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover preventOverflow={true}>Prevent overflow demo</Popover>;
+```
+
+### Reference Hidden
+
+Use the `referenceHidden` boolean to hide the Popover when the reference element is no longer visible in the viewport.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover referenceHidden={true}>Reference hidden demo</Popover>;
+```
+
+### Match Width
+
+Use the `matchWidth` boolean to match the width of the popover to the reference element.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover matchWidth={true}>Match width demo</Popover>;
+```
+
+### Role
+
+Use the `role` prop with `PopoverRole` enum to set the role of the popover.
+`PopoverRole.Dialog` if the content is interactive, or `PopoverRole.Tooltip` for purely informational popovers.
+
+Default: `PopoverRole.Tooltip`
+
+```jsx
+import { Popover, PopoverRole } from '../../ui/component-library';
+
+<Popover role={PopoverRole.Tooltip}>PopoverRole.Tooltip</Popover>;
+<Popover role={PopoverRole.Dialog}>PopoverRole.Dialog</Popover>;
+```
+
+### Offset
+
+Use the `offset` prop to pass an array of two numbers to offset the popover from the reference element.
+Default is `[0, 8]`
+First number controls the skidding offset and the second number controls the distance offset.
+
+```jsx
+import { Popover } from '../../ui/component-library';
+
+<Popover offset={[0, 32]}>offset override to [0,32]</Popover>;
+```
+
+### Esc Key Close
+
+`escKeyClose` is a callback function that is invoked when the 'Escape' key is pressed within the `Popover` component
```

Comment:
Should this be `onEscKeyClose`? cc @brianacnguyen what were our conventions and were are the docs 😝

Reply:
The correct naming convention should be either `onClose` or `onPressEscKey`